### PR TITLE
hqplayer-desktop: update tarball location and misc improvement

### PR DESCRIPTION
**Pull request by Et7f3**
###### Things done

I did not test because I have no use of this binary (and it is binary prebuilt). I grep-ed to validate path are all valid.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/hqplayer-desktop/default.nix [hunk 1]
```diff
@@ -1,11 +1,13 @@
 { mkDerivation
 , alsa-lib
 , autoPatchelfHook
+, evince
 , fetchurl
 , flac
 , gcc11
```
Comment (lovesegfault) [inline pkgs/applications/audio/hqplayer-desktop/default.nix:7]: You need to move to `gcc12` for the newer version

**Comment by lovesegfault** [comment]
Fixing `hqplayerd` may also be needed

File: pkgs/applications/audio/hqplayer-desktop/default.nix [hunk 2]
```diff
@@ -18,24 +20,25 @@
 
 mkDerivation rec {
   pname = "hqplayer-desktop";
-  version = "4.13.1-38";
+  version = "4.22.0-65";
```
Comment (Et7f3) [inline pkgs/applications/audio/hqplayer-desktop/default.nix:23]: Oops I didn't saw "HQPlayer 5 Desktop: 5.0.0" was released

**Comment by Et7f3** [comment]
> Fixing `hqplayerd` may also be needed

I can't find new version:

https://www.signalyst.eu/bins/naa/linux/fc36/networkaudiod-4.4.0-14.fc36.x86_64.rpm I guess networkaudiod is not hqplayerd (since version is lower)

**Pull request merged**